product_docs/docs/harp/2/01_release-notes.mdx

```diff
@@ -0,0 +1,178 @@
+---
+navTitle: Release notes
+title: Release notes
+---
+
+## Release 2.1.0 (2022-5-17)
+
+This is a minor release of HARP 2 that includes new features as well
+as fixes for issues identified in previous versions.
+
+### Enhancements
+
+-   **Leader per location**
+
+    Support for selecting a leader per location rather than relying on DCS
+    like etcd to have separate setup in different locations.
+
+    This still requires a majority of nodes to survive loss of a location,
+    so an odd number of both locations and database nodes is recommended.
+
+-   **Leader selection optimization for BDR DCS**
+
+    The BDR DCS now uses a push notification from the consensus rather than
+    through polling nodes. This change reduces the time for new leader selection and
+    the load that HARP does on the BDR DCS since it doesn't need to poll in
+    short intervals anymore.
+
+-   **Rolling updates**
+
+    TPA now restarts each HARP Proxy one by one and wait until they come back
+    to reduce any downtime incurred by the application during software
+    upgrades.
+
+-   **Embedded PGBouncer deprecation**
+
+    The support for embedding PGBouncer directly into HARP Proxy is now
+    deprecated and will be removed in the next major release of HARP.
+
+    It's now possible to configure TPA to put PGBouncer on the same node as
+    HARP Proxy and point to that HARP Proxy.
+
+
```

Comment:
### Bug fixes

- `harpctl promote <node_name>` would occasionally promote a different node than the one specified.  This has been fixed. (75406)

- `harpctl apply` no longer turns off routing for leader after the cluster has been established. (80790)

- Fencing would sometimes fail when using BDR as the Distributed Consensus Service.  This has been corrected.

- Harp-manager no longer exits if it cannot start a failed database. Harp-manager will keep retrying with randomly increasing periods. (78516)

- The internal pgbouncer proxy implementation had a memory leak.  This has been remediated.